src/Models/Solution.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace SolutionsService.Models
 {
-    public class Solution
+    public abstract class Solution
     {
         public Solution()
```

Comment:
C# classes do not inherit constructors. If this constructor is unused by the child classes it should be deleted. Else it should be protected instead of public.

Reply:
fixed in next commit